src/librbd/api/Image.cc

```diff
@@ -601,10 +601,24 @@ int Image<I>::remove(IoCtx& io_ctx, const std::string &image_name,
     ConfigProxy config(cct->_conf);
     Config<I>::apply_pool_overrides(io_ctx, &config);
 
+    std::string get_image_id;
+    int r = cls_client::dir_get_id(&io_ctx, RBD_DIRECTORY, image_name,
+                                   &get_image_id);
+    if (r < 0 && r != -ENOENT) {
+      lderr(cct) << "failed to retrieve image id: " << cpp_strerror(r) << dendl;
+      return r;
+    }
```

Comment:
`r == -ENOENT` would mean that is most likely v1 format image (or a damaged v2 image). In this case we should skip the trash move/remove exercise below and go to `librbd::image::RemoveRequest` part.

Another words, wrap the code below in `if (r == 0)` condition block.